### DISCUSSION TITLE
Unable to update sort key of defined GSI index

### DISCUSSION THREAD

**Discussion by FH-sandeepkaur1** [sensedeep/dynamodb-onetable#81] [General]
We are not able to update the sort key for GSI index using table or model api. 
Here is the explanation:

## Schema and index definition
```
schema: {
    indexes: {
        primary: { hash: 'pk', sort: 'sk' },
        'sample-gsi-index': {
            hash: 'pk',
            sort: 'stlStatus',
        },
    },
    models: {
        'settlement-model' : {
            pk: { type: String },
            sk: { type: String, value: 'STL#${id}' },
            id: { type: String, uuid: true, required: true },
            stlStatus: { type: String },
        }
    },
},
```

## Update query
```
await this.table.update('Settlements', {
          pk: 'xyz',
          sk: `STL#123`,
          stlStatus: 'updatedStatus',
      });
```

Ideally, stlStatus should be updated to updatedStatus but since it is a part of GSI definition index named sample-gsi-index as sort key. It is not able to update the value of it. When we remove the index definition itself then it updates the stlStatus value correctly.

**Comment by mobsense** [comment]
See Params.updateIndexes.

Because DynamoDB will create a new item if you ever accidentally change your keys on an update, for safety, we require updateIndexes to force an index change.

```

updateIndexes | boolean | Set to true to update index attributes. The default during updates is to not update index values which are defined during create.
-- | -- | --



````

**Reply by FH-sandeepkaur1** [reply]
I have updated my query with your solution and it is working as expected. 

Thank you
